### DISCUSSION TITLE
Setting up Phpactor as language server for PHP?

### DISCUSSION THREAD

**Discussion by superDuperCyberTechno** [helix-editor/helix#9665] [Q&A] [answered]
I need some help with setting up Phpactor for PHP.

I have the executable installed (`phpactor`) and the languages.toml file is set up and being parsed correctly. However, the following does not work:

```
[[language]]
name = "php"
language-servers = [ "phpactor" ]
```

Anyone know why?

**Comment by inf0dr0id** [comment, marked as answer]
https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#phpactor

**Comment by TornaxO7** [comment]
@superDuperCyberTechno got your question answered? If yes it would be nice if you could mark your question as answered :)

**Comment by xtrasmal** [comment]
@superDuperCyberTechno the manual won't help you much. I have it setup after a lot of fighting and will share when I get home.

**Reply by xtrasmal** [reply]
I'm not much help after all.. muhaha. I double checked and this is all needed. Helix comes with pretty good defaults

```toml
[[language]]
name = "php"
auto-format = true
language-servers = [ "phpactor"]

[language-server.phpactor]
command = "phpactor"
args = [ "language-server" ]
```
and a simple phpactor.json

```json
{
  "$schema": "phpactor.schema.json",
  "language_server_psalm.bin": "~/.config/composer/vendor/bin/psalm",
  "language_server_php_cs_fixer.bin": "~/.config/composer/vendor/bin/php-cs-fixer",
  "language_server_phpstan.bin": "~/.config/composer/vendor/bin/phpstan",
  "language_server_psalm.enabled": true,
  "language_server_php_cs_fixer.enabled": true,
  "language_server_phpstan.enabled": true
}   
```
works great at the moment.
